File: 0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const map = new Map()
    for (const str of strs) {
        const sortedChar = str.split("").sort().join("")
        if (map.has(sortedChar)) {
            map.get(sortedChar).push(str);
        } else {
            map.set(sortedChar, [str])
        }
    }
    const result = [];
    for (const value of map.values()) {
        result.push(value)
    }
    return result
};

File: 0049-group-anagrams/0049-group-anagrams.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map: MutableMap<String, MutableList<String>> = HashMap();
        for (i in 0 until strs.size) {
            val orginChar = strs.get(i);
            val sortedChar = orginChar.toCharArray().sorted().joinToString("")
            val value = map.getOrPut(sortedChar) { mutableListOf() }
            value.add(orginChar)
        }
        val result: MutableList<List<String>> = mutableListOf()
        for (value in map.values) {
            result.add(value)
        }
        return result;
    }
}